porch/pkg/registry/porch/strategy.go

```diff
@@ -67,9 +68,16 @@ func (s packageRevisionStrategy) ValidateUpdate(ctx context.Context, obj, old ru
 				}, ",")),
 			))
 		}
-	case api.PackageRevisionLifecyclePublished:
+	case api.PackageRevisionLifecyclePublished, api.PackageRevisionLifecycleDeletionProposed:
 		// We don't allow any updates to the spec for packagerevision that have been published. That includes updates of the lifecycle. But
-		// we allow updates to metadata and status.
+		// we allow updates to metadata and status. The only exception is that the lifecycle
+		// can change between Published and DeletionProposed.
```

Comment:
We probably also should allow the opposite move. So a packagerevision can transition from deletionProposed back to published.

Reply:
Ah yeah this PR does allow the opposite move - I'll update the comment to reflect that